packages/integrations/useIDBKeyval/index.ts

```diff
@@ -64,16 +74,24 @@ export function useIDBKeyval<T>(
   const data = (shallow ? shallowRef : deepRef)(initialValue) as Ref<T>
 
   const rawInit: T = toValue(initialValue)
+  const serializer = options.serializer
 
   async function read() {
     try {
       const rawValue = await get<T>(key)
       if (rawValue === undefined) {
-        if (rawInit !== undefined && rawInit !== null && writeDefaults)
-          await set(key, rawInit)
+        if (rawInit !== undefined && rawInit !== null && writeDefaults) {
+          const initValue = serializer ? serializer.write(rawInit) : rawInit
+          await set(key, initValue)
+        }
       }
       else {
-        data.value = rawValue
+        if (serializer) {
+          data.value = serializer.read(rawValue)
+        }
+        else {
+          data.value = rawValue
+        }
```

Comment:
maybe we should do this via default serializer. Like:

```suggestion
        data.value = serializer.read(rawValue)
```

and

```ts
  const {
    flush = 'pre',
    deep = true,
    shallow = false,
    onError = (e) => {
      console.error(e)
    },
    writeDefaults = true,
    serializer: {
      read: (val)=>val,
      write: (val)=>val, 
    }
  } = options
```